src/session/client-certificate.c

```diff
@@ -300,8 +368,39 @@ cockpit_session_client_certificate_map_user (const char *client_certificate_file
       return NULL;
     }
 
+  /* check if stdin is a Unix socket; then we are being called via cockpit-session@.service and need to
+   * check the cgroup of our peer */
+  struct ucred ucred;
+  int ws_pid_dirfd;
+  if (getsockopt (STDIN_FILENO, SOL_SOCKET, SO_PEERCRED, &ucred, &(socklen_t){sizeof ucred}) == 0)
```

Comment:
Are we supposed to verify the value of `optlen` after the call?

Reply:
https://man7.org/linux/man-pages/man2/getsockopt.2.html doesn't mention this specifically, but let's not take chances. I added an assertion.